### ISSUE TITLE
Mention in docs "Strip Linked Product" needs to be turned off for TestFlight

### ISSUE THREAD

**Issue by avishayil** [realm/realm-js#311]
Seems that for some reason the app can't find RealmReact framework, even though it is included.
When I run-ios on the simulator everything works great, and when I'm running on device (without TestFlight) everything works great too.

I tried to install via TestFlight, didn't try production of course.

iOS 9.2.1, React Native 0.21, Realm 10.0

Stacktrace:
````
Mar  6 13:04:48 Avishay-Bassas-iPhone amfid[224] <Error>:  SecTrustEvaluate  [leaf IssuerCommonName SubjectCommonName]
Mar  6 13:04:48 Avishay-Bassas-iPhone Geektime[421] <Warning>: Failed to load RCTRegisterModule symbol - 
Mar  6 13:04:48 Avishay-Bassas-iPhone Geektime[421] <Warning>: [Crashlytics] Version 3.6.0 (99)
Mar  6 13:04:49 Avishay-Bassas-iPhone Geektime[421] <Error>: Missing Realm constructor - please ensure RealmReact framework is included!
Mar  6 13:04:49 Avishay-Bassas-iPhone Geektime[421] <Critical>: Unhandled JS Exception: Missing Realm constructor - please ensure RealmReact framework is included!
Mar  6 13:04:49 Avishay-Bassas-iPhone Geektime[421] <Error>: Module AppRegistry is not a registered callable module.
Mar  6 13:04:49 Avishay-Bassas-iPhone Geektime[421] <Error>: *** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: Missing Realm constructor - please ensure RealmReact framework is included!', reason: 'Unhandled JS Exception: Missing Realm constructor - please ensure RealmReac..., stack:
	<unknown>@25:8658
	r@1:617
	n@1:283
	<unknown>@25:7982
	r@1:617
	n@1:283
	<unknown>@21:7356
	r@1:617
	n@1:283
	<unknown>@21:2889
	r@1:617
	n@1:283
	<unknown>@6:31488
	r@1:617
	r@1:346
	n@1:283
	<unknown>@29:470
	'
	*** First throw call stack:
	(0x25efe2eb 0x256cadff 0x25efe231 0x619af 0x5cf43 0x25f02cc4 0x25e2d2ad 0x25e30d47 0x5fc41 0x7bf51 0x7ba23 0x25a9bdd7 0x25aa5f97 0x25a9ef2f 0x25aa77b9 0x25aa7193 0x25c34e0d 0x25c349fc)
Mar  6 13:04:49 Avishay-Bassas-iPhone SpringBoard[48] <Warning>: HW kbd: Failed to set (null) as keyboard focus
Mar  6 13:04:49 Avishay-Bassas-iPhone SpringBoard[48] <Warning>: UNNotificationRegistrarConnectionListener connection invalidated
Mar  6 13:04:49 Avishay-Bassas-iPhone com.apple.xpc.launchd[1] (UIKitApplication:com.zuznow.geektime[0x44f7][421]) <Notice>: Service exited due to signal: Abort trap: 6
Mar  6 13:04:49 Avishay-Bassas-iPhone SpringBoard[48] <Warning>: Application 'UIKitApplication:com.zuznow.geektime[0x44f7]' crashed.
Mar  6 13:04:49 Avishay-Bassas-iPhone UserEventAgent[23] <Warning>: 361724818041: id=com.zuznow.geektime pid=421, state=0
Mar  6 13:04:50 Avishay-Bassas-iPhone diagnosticd[82] <Error>: assertion failed: 13D15: diagnosticd + 10531 [BBA2CF30-B9BF-328F-8EF8-F8662660E017]: 0x1
Mar  6 13:04:50 Avishay-Bassas-iPhone Unknown[82] <Error>: 
Mar  6 13:04:50 Avishay-Bassas-iPhone diagnosticd[82] <Error>: Invalid trace slot 0 location 6221184 for image 'E0D5FDCF-86A9-3291-9799-E68A99F2773A'
Mar  6 13:04:50 Avishay-Bassas-iPhone diagnosticd[82] <Error>: assertion failed: 13D15: diagnosticd + 10531 [BBA2CF30-B9BF-328F-8EF8-F8662660E017]: 0x1
Mar  6 13:04:50 Avishay-Bassas-iPhone Unknown[82] <Error>: 
Mar  6 13:04:50 Avishay-Bassas-iPhone diagnosticd[82] <Error>: Invalid trace slot 0 location 6221231 for image 'E0D5FDCF-86A9-3291-9799-E68A99F2773A'
Mar  6 13:04:50 Avishay-Bassas-iPhone diagnosticd[82] <Error>: assertion failed: 13D15: diagnosticd + 10531 [BBA2CF30-B9BF-328F-8EF8-F8662660E017]: 0x1
Mar  6 13:04:50 Avishay-Bassas-iPhone Unknown[82] <Error>: 
Mar  6 13:04:50 Avishay-Bassas-iPhone diagnosticd[82] <Error>: assertion failed: 13D15: diagnosticd + 10531 [BBA2CF30-B9BF-328F-8EF8-F8662660E017]: 0x1
Mar  6 13:04:50 Avishay-Bassas-iPhone diagnosticd[82] <Error>: Invalid trace slot 0 location 13528788 for image 'E0D5FDCF-86A9-3291-9799-E68A99F2773A'
Mar  6 13:04:50 Avishay-Bassas-iPhone Unknown[82] <Error>: 
Mar  6 13:04:50 Avishay-Bassas-iPhone diagnosticd[82] <Error>: assertion failed: 13D15: diagnosticd + 10531 [BBA2CF30-B9BF-328F-8EF8-F8662660E017]: 0x1
Mar  6 13:04:50 Avishay-Bassas-iPhone diagnosticd[82] <Error>: Invalid trace slot 0 location 6221184 for image 'E0D5FDCF-86A9-3291-9799-E68A99F2773A'
Mar  6 13:04:50 Avishay-Bassas-iPhone Unknown[82] <Error>: 
Mar  6 13:04:50 Avishay-Bassas-iPhone diagnosticd[82] <Error>: Invalid trace slot 0 location 6221231 for image 'E0D5FDCF-86A9-3291-9799-E68A99F2773A'
Mar  6 13:04:50 Avishay-Bassas-iPhone ReportCrash[422] <Notice>: platform_thread_get_unique_id matched 59745
Mar  6 13:04:50 Avishay-Bassas-iPhone ReportCrash[422] <Notice>: Formulating report for corpse[421] Geektime
Mar  6 13:04:50 Avishay-Bassas-iPhone searchd[179] <Warning>: ====^^^^ DuetExpert missing data, count -> people:1 applicationDeepLinks:0 applications:8 requests:31 missingAllDataRequests:0
Mar  6 13:04:50 Avishay-Bassas-iPhone ReportCrash[422] <Warning>: saved type '109_Geektime' report (1 of max 25) as /var/mobile/Library/Logs/CrashReporter/Geektime_2016-03-06-130449_Avishay-Bassas-iPhone.ips.beta
Mar  6 13:04:51 Avishay-Bassas-iPhone SpringBoard[48] <Warning>: ADDING REMOTE com.zuznow.geektime, <BBRemoteDataProvider 0x16b561d0; com.zuznow.geektime>
Mar  6 13:04:52 Avishay-Bassas-iPhone SpringBoard[48] <Warning>: dataProviderDidLoad: com.zuznow.geektime
Mar  6 13:05:00 Avishay-Bassas-iPhone OTACrashCopier[423] <Notice>: (Warn ) <OTACrashCopierController.m __42-[OTACrashCopierController handleRequest:]_block_invoke:571> Creating Power Assertion for 600 secs
Mar  6 13:05:00 Avishay-Bassas-iPhone OTACrashCopier[423] <Notice>: (Warn ) <OTALogSubmissionOperation.m -[OTALogSubmissionOperation postToServer:forRouting:ofLogs:withConnection:usingParams:]:152> 'job-1-beta' sending request of 14093 bytes
Mar  6 13:05:11 Avishay-Bassas-iPhone OTACrashCopier[423] <Warning>: Scheduling on exit
Mar  6 13:05:18 Avishay-Bassas-iPhone launchd[1] <Notice>: BUG in libdispatch: 13D15 - 1631 - 0x0
Mar  6 13:05:18 Avishay-Bassas-iPhone networkd[89] <Notice>: BUG in libdispatch: 13D15 - 1631 - 0x0
Mar  6 13:05:18 Avishay-Bassas-iPhone syncdefaultsd[397] <Notice>: (Note ) marked "com.me.keyvalueservice" topic as "enabled" on <APSConnection: 0x14d91180>
Mar  6 13:05:18 Avishay-Bassas-iPhone apsd[96] <Notice>: BUG in libdispatch: 13D15 - 1631 - 0x0
````

**Comment by avishayil** [comment]
Apparently, found the same issue here:
https://github.com/realm/realm-js/issues/219#issuecomment-178066140

Checking if it solves my issue.
This should really be added to the documentation...

**Comment by avishayil** [comment]
Solved it.

**Comment by appden** [comment]
@avishayil Sorry you ran into that! We will add it to the docs. It doesn't appear there's a good way for us to avoid this step, but we will keep searching.

**Comment by avishayil** [comment]
Great! Thanks.

**Comment by avishayil** [comment]
What about production builds? Can someone confirm if this step is needed there?

**Comment by esutton** [comment]
This issue should be renamed.  It does not seem to have anything specific to do with **TestFlight** deployment.

I am using react-native and realm on iOS.  I had same crash deploying an ipa package using my local iTunes client. 

select your **Target > Deployment**

- **Strip Debug Symbols During Copy: No**
- **Strip Linked Product: No**
 
Thank you @avishayil

**Comment by jwhitley** [comment]
Using `0.11.0-rc` out of npm, I appear to be seeing this issue when deploying either via TestFlight or directly loading an .ipa (Archive, then Export) to a device via Xcode.  The primary symptom is the hallmark `Missing Realm constructor - please ensure RealmReact.framework is included!` exception.

This repros with both RN 0.21.0 and 0.22.1.

This app runs fine both in the simulator and on device in normal development workflow, and Realm is clearly persisting data across sessions in my app.

I have absolutely, definitely triple-checked the aforementioned deployment settings are set as:

* **Strip Debug Symbols During Copy: No**
* **Strip Linked Product: No**

**Comment by alazier** [comment]
@jwhitley 

Looks like you also need to set **Dead Code Stripping: No**

**Comment by jwhitley** [comment]
@alazier I had Dead Code Stripping set to No on the target, but realized that I'd missed this setting on the project.  Corrected that, then burned the world: removed/reinstalled `node_modules`, cleaned the project, rebuilt the archive, same result.  :neutral_face: 

I'm going to see if I can reproduce this on my simple public test repo.

**Comment by esutton** [comment]
This problem came back for me.  I thought setting 2 strip options had fixed it.
- 
- iOS 9.3, 
- react-native": "^0.20.0",
- realm": "^0.10.0",

```
Mar 22 13:52:52 esutton-ipad-air SpringBoard[54] <Error>:  SecTrustEvaluate  [leaf IssuerCommonName SubjectCommonName]
Mar 22 13:52:53 esutton-ipad-air kernel[0] <Notice>: xpcproxy[940] Container: /private/var/mobile/Containers/Data/Application/BA5202F1-8212-4666-9DA9-E79680DB0EEF (sandbox)
Mar 22 13:52:52 esutton-ipad-air FieldPlanner[940] <Warning>: Failed to load RCTRegisterModule symbol - 
Mar 22 13:52:53 esutton-ipad-air FieldPlanner[940] <Error>: Missing Realm constructor - please ensure RealmReact framework is included!
Mar 22 13:52:53 esutton-ipad-air FieldPlanner[940] <Critical>: Unhandled JS Exception: Missing Realm constructor - please ensure RealmReact framework is included!
Mar 22 13:52:53 esutton-ipad-air FieldPlanner[940] <Error>: *** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: Missing Realm constructor - please ensure RealmReact framework is included!', reason: 'Unhandled JS Exception: Missing Realm constructor - please ensure RealmReac..., stack:
	<unknown>@29:8533
	r@1:617
	n@1:283
```

**Comment by appden** [comment]
As mentioned in #345, as of Xcode 7.3, projects also need to disable bitcode by setting `Enable Bitcode` to `No` in the Build Settings. I am working on a solution where we will not need to use all of these custom Xcode settings.

**Comment by appden** [comment]
@jwhitley @esutton @avishayil We just released 0.11.0, which switches to using a static library rather than a framework. Check out the [0.11.0 Release Notes](https://github.com/realm/realm-js/releases/tag/v0.11.0) for how to upgrade your project. You can revert your strip and bitcode settings to the defaults after upgrading.

**Comment by jwhitley** [comment]
@appden, confirmed the new build process in the release notes working on device in both development  and release builds using 0.11.0.  Brilliant, thanks!
